compiler/semdata.nim

```diff
@@ -103,7 +103,7 @@ type
     imports*: seq[ImportedModule] # scope for all imported symbols
     topLevelScope*: PScope     # scope for all top-level symbols
     p*: PProcCon               # procedure context
-    intTypeCache*: array[-5..32, PType] # cache some common integer types
+    intTypeCache*: array[27, PType] # cache some common integer types
```

Comment:
This is an actual use case for negative indexed arrays. We map values literally from the integers `-5, -4, -3, ..., 30, 31, 32` to type values.

You could either make this `0 .. 32` and lose the optimization for -5 .. -1, or you could set the length to be `32 - (-5)` (which is not 27) and then manually write `-5` for `low(intTypeCache)` and manually do `- (-5)` when doing the index access. If only we had a way to express this with the type system